payment-service/src/main/java/by/sergo/paymentservice/service/impl/CreditCardServiceImpl.java

```diff
@@ -0,0 +1,132 @@
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+
+import static by.sergo.paymentservice.domain.enums.Operation.PAYMENT;
+
+@Service
+@RequiredArgsConstructor
+@Transactional(readOnly = true)
+public class CreditCardServiceImpl implements CreditCardService {
+    private final CreditCardRepository creditCardRepository;
+    private final AccountRepository accountRepository;
+    private final TransactionStoreRepository transactionStoreRepository;
+    private final CreditCardMapper creditCardMapper;
+
+    @Override
+    @Transactional
+    public CreditCardResponse addCard(CreditCardCreateUpdate dto) {
+        checkUniqueCreditCardAndExpirationDate(dto);
+        var creditCard = creditCardMapper.mapToEntity(dto);
+        creditCard.setId(null);
+        var savedCreditCard = creditCardRepository.save(creditCard);
+        return creditCardMapper.mapToDto(savedCreditCard);
+    }
+
+    @Override
+    @Transactional
+    public CreditCardResponse changeCard(CreditCardCreateUpdate dto, Long id) {
+        checkCardIsUniqueForUpdate(dto, id);
+        var creditCard = creditCardMapper.mapToEntity(dto);
+        creditCard.setId(id);
+        return creditCardMapper.mapToDto(creditCardRepository.save(creditCard));
+    }
+
+    @Override
+    @Transactional
+    public CreditCardResponse deleteById(Long id) {
+        var creditCard = getByIdOrElseThrow(id);
+        creditCardRepository.deleteById(id);
+        return creditCardMapper.mapToDto(creditCard);
+    }
+
+    @Override
+    public CreditCardResponse getDriverCard(Long driverId) {
+        var creditCard = creditCardRepository.findByUserIdAndUserType(driverId, UserType.DRIVER)
+                .orElseThrow(() -> new NotFoundException(ExceptionMessageUtil.getNotFoundMessage("Credit card with type driver", "userId", driverId)));
+        return creditCardMapper.mapToDto(creditCard);
+    }
+
+    @Override
+    public CreditCardResponse getPassengerCard(Long passengerId) {
+        CreditCard creditCard = getPassengerCreditCard(passengerId);
+        return creditCardMapper.mapToDto(creditCard);
+    }
+
+    @Override
+    @Transactional
+    public CreditCardResponse makePayment(PaymentRequest payment) {
+        var passengerCreditCard = getPassengerCreditCard(payment.getPassengerId());
+        var driverAccount = accountRepository.findByDriverId(payment.getDriverId())
+                .orElseThrow(() -> new NotFoundException(ExceptionMessageUtil.getNotFoundMessage("Account", "driverId", payment.getDriverId())));
+
+        if (passengerCreditCard.getBalance().compareTo(payment.getSum()) >= 0) {
+            passengerCreditCard.setBalance(passengerCreditCard.getBalance().subtract(payment.getSum()));
+        } else {
+            throw new BadRequestException(ExceptionMessageUtil.getWithdrawalExceptionMessage("Credit card", "passengerId", payment.getPassengerId().toString(), passengerCreditCard.getBalance().toString()));
+        }
```

Comment:
Можно сделать без else
```suggestion
        if (passengerCreditCard.getBalance().compareTo(payment.getSum()) < 0) {
            throw new BadRequestException(ExceptionMessageUtil.getWithdrawalExceptionMessage("Credit card", "passengerId", payment.getPassengerId().toString(), passengerCreditCard.getBalance().toString()));
        } 
        
        passengerCreditCard.setBalance(passengerCreditCard.getBalance().subtract(payment.getSum()));
```